src/Services/FileLoader/CommonMarkExtensions/HeadingWrapper.php

```diff
@@ -0,0 +1,92 @@
+<?php
+
+
+namespace App\Services\FileLoader\CommonMarkExtensions;
+
+
+use League\CommonMark\Block\Element\AbstractBlock;
+use League\CommonMark\Block\Element\Heading;
+use League\CommonMark\Block\Renderer\BlockRendererInterface;
+use League\CommonMark\ElementRendererInterface;
+use League\CommonMark\HtmlElement;
+
+class HeadingWrapper implements BlockRendererInterface
+{
+    private $headers = [];
+
+    public function render(AbstractBlock $block, ElementRendererInterface $htmlRenderer, $inTightList = false)
+    {
+        $render = $htmlRenderer->renderInlines($block->children());
+        $pureTitle = strip_tags($render);
+
+        if ($block instanceof Heading) {
+            $this->headers[] = ["text" => $pureTitle, "level" => $level = $block->getLevel(), "id" => $id = "h$level-" . $this->slugify($pureTitle) . "-" . substr(md5($render), 0, 10)];
```

Comment:
If possible, indenting theses long multi parameters string in vertical to be more readable.
```php
["text" => $pureTitle,
 "level" => $level = $block->getLevel(),
...,
]